### PR TITLE
Feature/Create the KRR database schema

### PR DESCRIPTION
**Pull request by Ahmed-Ghanam**
## Description
Establish three tables to store contact and reservation register data.

## Related Issue(s)
- #186 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Comment by Ahmed-Ghanam** [comment]
This PR is built on top of [Reorganize Profile solution to a more clean architecture](https://github.com/Altinn/altinn-profile/pull/199)

File: src/Altinn.Profile.Integrations/Migration/profiledb.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    ENCODING = 'UTF8'
+    LC_COLLATE = 'Norwegian_Norway.1252'
+    LC_CTYPE = 'Norwegian_Norway.1252'
+    LOCALE_PROVIDER = 'libc'
+    TABLESPACE = pg_default
+    CONNECTION LIMIT = -1
+    IS_TEMPLATE = False;
+
+-- Create schema if it doesn't exist
+CREATE SCHEMA IF NOT EXISTS contact_and_reservation;
```
Comment (SandGrainOne) [inline src/Altinn.Profile.Integrations/Migration/profiledb.sql:17]: We need statements that grants the two users we have access to the schema:

GRANT ALL ON SCHEMA contact_and_reservation TO platform_profile_admin;
GRANT USAGE ON SCHEMA contact_and_reservation TO platform_profile;

File: src/Altinn.Profile.Integrations/Migration/profiledb.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿-- Drop the database if it exists
+DROP DATABASE IF EXISTS profiledb;
```
Comment (SandGrainOne) [inline src/Altinn.Profile.Integrations/Migration/profiledb.sql:2]: Remove this line and the statement for creating the database. The database is created for us with terraform.

File: src/Altinn.Profile.Integrations/Migration/profiledb.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    mailbox_address VARCHAR(50),
+    mailbox_supplier_id_fk INT,
+    x509_certificate TEXT,
+    language_code CHAR(2) NULL,
+    CONSTRAINT fk_mailbox_supplier FOREIGN KEY (mailbox_supplier_id_fk) REFERENCES contact_and_reservation.mailbox_supplier (mailbox_supplier_id),
+    CONSTRAINT chk_language_code CHECK (language_code ~* '^[a-z]{2}$')
+);
+
+-- Indexes for performance
+CREATE INDEX idx_mailbox_supplier_id_fk ON contact_and_reservation.person (mailbox_supplier_id_fk);
```
Comment (SandGrainOne) [inline src/Altinn.Profile.Integrations/Migration/profiledb.sql:53]: This index is probably unnecessary. I don't see a scenario where we will search for persons based on mailbox supplier.

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**